addon/index.js

```diff
@@ -18,15 +18,32 @@ export default class Resolver {
 
   moduleNameForFullName(fullName) {
     // TODO: development assertion or warning if not already normalized
-    let index = fullName.indexOf(':');
-    let type = fullName.substring(0, index)
-    let name = fullName.substring(index + 1, fullName.length);
-    let moduleName = this.namespace.modulePrefix + '/';
+    let prefix, type, name, moduleName;
+
+    const fullNameParts = fullName.split('@');
+
+    // HTMLBars uses helper:@content-helper which collides
+    // with ember-cli namespace detection.
+    // This will be removed in a future release of HTMLBars.
+    if (fullNameParts.length === 2 && fullName !== 'helper:@content-helper') {
+      const prefixParts = fullNameParts[0].split(':');
+
+      if (prefixParts.length === 2) {
+        [type, prefix] = prefixParts;
```

Comment:
this wont transpile nicely, as the compiler is unaware if `prefixParts` is an array or an iterable.